wvd-templates/DSC/Functions.ps1

```diff
@@ -222,13 +222,159 @@ function ExtractDeploymentAgentZipFile {
     
     # Locating and extracting DeployAgent.zip
     Write-Log -Message "Locating DeployAgent.zip within Custom Script Extension folder structure: $ScriptPath"
-    $DeployAgentFromRepo = (Get-ChildItem $ScriptPath\ -Filter DeployAgent.zip -Recurse | Select-Object).FullName
+    $DeployAgentFromRepo = (Get-ChildItem $ScriptPath\ -Filter DeployAgent.zip -Recurse | Select-Object -First 1).FullName

```

Comment:
Instead of only selecting the first item it seems like this should throw an exception if more than one "DeployAgent.zip" is found since that is not expected.

Reply:
@lintfan
I'm not following your comment. Can you please clarify ?

Reply:
@lintFan 
Are you just stating what it would do if first 1 was not selected (original code) and there were multiple files matched ? Or are you stating the new change would cause an exception ?

Reply:
I am suggesting a new change that would throw an exception rather than only selecting the first item.
For example if I have a directory structure like:
E:\git\test\
    target.txt
    test.ps1
    \sub 
        target.txt
Running (Get-ChildItem E:\git\test\ -Filter target.txt -Recurse | Select-Object) will return entries for both target.txt files. But with the directory structure of this code I can't think of a situation where multiple DeployAgent.zip files should be expected to be found within $ScriptPath.

Reply:
You are right that there should always be just 1 file. But just in case there might be others, we should handle it. I agree that we should throw an error if there's more than 1, but before this change, if there's more than 1, expand-archive would throw the error which would confuse the user. So I'll add another check to make sure there's only one file match